docs/design/deviceshifu/design-deviceShifu-rtsp.md

```diff
@@ -0,0 +1,122 @@
+# deviceShifu RTSP overall design
+
+deviceShifu-RTSP allows shifu connect to rtsp server and save the video into file. User can watch rtsp video using media player like VLC and get history video file from PV.
+
+## Goal
+
+Create a deviceShifu-RTSP using third-party library and allow user to connect RTSP server, and record the rtsp stream into file like mp4 / flv and so on, user can export file out of k8s.
+
+## General Design
+
+deviceShifu-RTSP will receive RESTful style request like other, then transfer the request into RTSP method calls and send it to RTSP Server, meanwhile deviceShifu-RTSP provide a RTSP address which forward the RTSP stream from the target RTSP server.
```

Comment:
Are we hosting RTSP server as well? Should we allow it push to user provided RTSP server?

Reply:
I think we need to host a RTSP Server for if we use the `redirect` of RTSP , the target server like telemetry service not expose from the kubernetes, the RTSP device cannot forward its stream to target server, so that we need host a RTSP Server to forward it